src/hammer_config/config_src.py

```diff
@@ -278,6 +280,68 @@ def subst_rename(key: str, value: Any, target_setting: str, replacement_setting:
                                         target_settings=subst_targets,
                                         rename_target=subst_rename)
 
+    def trysubst_str(input_str: str, replacement_func: Callable[[str], str]) -> str:
+        """Substitute ${...}"""
+
+        # this check allows a reference like "${foo}" to just be replaced
+        # instead of using regexes. therefore substituting int/bool/null works
```

Comment:
How does this differ from crossref?

If `"abc ${foo}"` is `trysubst`ed, how does it work if `foo` is an int or bool?

Reply:
it breaks. i found no good alternative there. 

but the common case is when a key refers to "${foo}", I want this to work regardless of what the type of the setting it.

the only difference between trysubst and subst, is in the subst_target() function: 
the trysubst will simply return no targets if it detects a non str/List[str] setting value, which is what I want. I left the default behaviour of subst alone.

I would prefer to just have subst, and change the behavior to allow non-string setting values instead of raising an exception.